translationUtils.ts

```diff
@@ -0,0 +1,284 @@
+import fetch from 'node-fetch';
+import imgur from 'imgur';
+import { Image, createCanvas } from 'canvas';
+
+imgur.setClientId('[REDACTED]');
+// This might not be necessary according to the docs, but can't hurt.
+imgur.setCredentials('[REDACTED]', '[REDACTED]', '[REDACTED]');
+
+type ImageBase64 = string; // Eg 'data:image/png;base64,iVBORw0KGgo...'
+
+interface TranslationResult {
+	originalLanguage: string;
+	translatedText: string;
+	minX: number;
+	minY: number;
+	maxX: number;
+	maxY: number;
+}
+
+enum StatusCode {
+	Ok = 200
+}
+
+// Returns null if the image could not be uploaded.
+export async function uploadImage(image: ImageBase64): Promise<URL | null> {
+	const formattedBase64 = image.replace(/^data:.*;base64,/g, '');
+	return new Promise(resolve =>
+		imgur
+			.uploadBase64(formattedBase64)
+			.then(function (json) {
+				const url = tryParseUrl(json.data.link);
+				resolve(url);
+			})
+			.catch(function () {
+				// Imgur can be down due to maintenance, high load, or other reasons.
+				// It is not an exceptional event, so simply return null.
+				resolve(null);
+			})
+	);
+}
+
+// Returns null if url is malformed.
+export function tryParseUrl(url: string): URL | null {
+	try {
+		return new URL(url);
+	} catch {
+		return null;
+	}
+}
+
+// Returns null if the image cannot be translated.
+export async function translate(image: ImageBase64): Promise<ImageBase64 | null> {
+	const translationRequest = await fetch('http://localhost:3001/translate', {
+		method: 'POST',
+		headers: {
+			'Content-Type': 'application/json'
+		},
+		// API supports multiple images, but only one is necessary for now.
+		body: JSON.stringify({ base64Images: [image], apiKey: '[REDACTED]' })
+	});
+
+	if (translationRequest.status !== StatusCode.Ok) {
+		throw new Error(`translation request failed with:\n${JSON.stringify(translationRequest)}`);
```

Comment:
Note the inclusion of parameters in the error message. Help your future self debug by including what runtime values the function went wrong with.